net/mwan3/files/usr/sbin/mwan3track

```diff
@@ -109,23 +117,58 @@ main() {
 			if [ $host_up_count -lt $reliability ]; then
 				case "$track_method" in
 					ping)
-						ping -I $DEVICE -c $count -W $timeout -s $size -q $track_ip &> /dev/null ;;
+							if [ $check_quality -eq 0 ]; then
+								ping -I $DEVICE -c $count -W $timeout -s $size -q $track_ip &> /dev/null
```

Comment:
@codemarauder during testing i have noticed a problem!

The check in line 138 is always success and system will not detect a fault!
On my system if apply your changes and use only ping without quality check the ping check is always successful even though the target ping could not get reached!

Do fix this we have to save the **$?** into a variable for example **$result** on **ping/arping/httping** and then change the check from **if [ $? -eq 0 ]; then** to **if [ $result -eq 0 ]; then**
this should fix the problem.

Reply:
Got it!
I recall why I had to use $ping_result, as $? was always true for me as well. I somehow missed that other methods are also using $? and need to be modified.

I will try to fix things tonight.